packages/lib-classifier/src/hooks/useFreehandLineReductions.js

```diff
@@ -4,45 +4,50 @@ import { useCaesarReductions } from './'
 
 export default function useFreehandLineReductions() {
   const {
-    subjectViewer: {
-      frame
-    },
     workflows: {
       active: workflow
     },
     workflowSteps: {
       active: step,
       findTasksByType
+    },
+    subjects: {
+      active: activeSubject,
     }
   } = useStores()
 
   const { loaded, caesarReductions } = useCaesarReductions(workflow.caesarReducer)
+  const stepIndex = parseInt(step.stepKey.replace('S',''), 10)
+
+  if (loaded
+    && getType(caesarReductions).name === 'FreehandLineReductions'
+    && activeSubject.caesarReductionsLoadedForStep[stepIndex] !== true) {
+    
+    // step.tasks is only the CURRENT TASK
+    // Test project with 3 steps means we need to run this 3 times to check for reductions on each step
+    // This is likely due to the potentially recursive nature of step navigation
 
-  if (loaded && getType(caesarReductions).name === 'FreehandLineReductions') {
     const caesarMarks = caesarReductions.findCurrentTaskMarks({
       stepKey: step.stepKey,
-      tasks: step.tasks,
-      frame,
+      task: step.tasks[0],
```

Comment:
The changes in this file are still a bit confusing to me. The purpose of `useFreehandLineReductions()` is to fetch reductions via `useCaesarReductions()`, then filter the reductions for marks that belong to the current `step`. Is that the correct understanding? Or are the changes in this file intended to filter marks per `task` too, and why?

I think my confusion stems from knowing that **in the FEM mobx store**, each workflow has a map of steps (`workflow.steps`). Each `step` has an array of `taskKeys`, which maps to the workflow's `tasks` array. Essentially, tasks are grouped into steps, and having more than one of either is optional. There's further explanation in [ADR 5](https://github.com/zooniverse/front-end-monorepo/blob/master/docs/arch/adr-05.md), and ongoing work in the fem-lab to build the corresponding UI.

This question might help clarify too: Why is FreehandLineReductions.js's `findCurrentTaskMarks` concerned with the `task.type`? My understanding is that the FreehandLineReductions mobx store model is relevant when `reductions` fetched via `useCaesarReductions()` hook have a `reducer` that equals 'machineLearnt'. Why does `reductions.data` need to be filtered based on task type, tool type, etc?

Apologies for harping on this topic 😆 but because there are several changes in this PR to freehand line code files, I want to be sure a clear understanding is documented.

Reply:
The `useFreehandLineReductions()` fetches reductions via `useCaesarReductions()` that get loaded per subject. They then were filtered for the current `step`, `task`, and `tool`. This was meant to ensure that's what the researcher wanted and is verifiable by the caesar data. 

That said, as per your comment above with the "contrived example", I don't think that should be necessary. I think the constraint can instead be on the `step` it gets loaded and that's it. This is a good example to bring up to Cliff about how we load caesar data across the board and may be part of an ADR of some kind.